### PR TITLE
Bug fix/rpm install and upgrade

### PR DESCRIPTION
**Pull request by fceller**
https://github.com/arangodb/planning/issues/4277

### CODE REVIEW

**Comment by fceller** [comment]
Please note that this will only fix upgrade FROM packages created with this fix. Upgrades from older package will still not work.

**Comment by fceller** [comment]
Needs to be tested

**Comment by KVS85** [comment]
Tried with CentOS 7. Built 4 RPM packages (changing CMakeLists.txt and perform subset of makeRelease function commands of oskar):
- based on 3.4 branch: 3.4.8(-1), 3.4.9(-1) 
- based on 3.5 branch: 3.5.0(-1), 3.5.1(-1)

Then `sudo rpm -i <3.4.8(-1)>.rpm`, set it up (changed server password, started service, connected and made some entities in DB), `sudo rpm -i <3.4.9(-1)>.rpm` finished well, but service is not sarted and `sudo systemctl start arangodb3.service` says `Failed to start arangodb3.service: unit not found`.

For separated 3.4.9(-1) to 3.5.0(-1) and 3.5.0(-1) to 3.5.1(-1) checks it's the same.

File: rpm/3.4/arangodb3e.spec.in [hunk 1]
```diff
@@ -232,13 +235,13 @@ exit 0
...
 if ! getent group arangodb >/dev/null 2>&1; then
-        /usr/sbin/groupadd -r arangodb
+  /usr/sbin/groupadd -r arangodb
 fi
 
 # create arangodb user only if it doesn't already exist
 if ! getent passwd arangodb >/dev/null 2>&1; then
-        /usr/sbin/useradd -r -g arangodb --home %{_localstatedir}/arangodb arangodb
-        /usr/sbin/usermod -c "Arango Server" arangodb
+  /usr/sbin/useradd -r -g arangodb --home %{_localstatedir}/arangodb arangodb
```
Comment (dothebart) [inline rpm/3.4/arangodb3e.spec.in]: are the RPM distributions also moving into the direction of putting everything to one directory, as we saw with the debian docker containers? Or should we first have a `test -f` and else to `/sbin/chmod ` etc?

File: rpm/3.4/arangodb3e.spec.in [hunk 2]
```diff
@@ -257,20 +260,26 @@ else
...
+    systemctl daemon-reload
+  else
+    chkconfig --level 2345 arangodb3 off
+    rm %{_sysconfdir}/init.d/arangodb3
+  fi
+  rm %{_sysconfdir}/logrotate.d/arangodb3
 fi
-rm %{_sysconfdir}/logrotate.d/arangodb3
+
+exit 0
```
Comment (dothebart) [inline rpm/3.4/arangodb3e.spec.in]: shouldn't this rather be the else case? Or was this exit missing previously like in all the other cases?

Comment (fceller) [reply rpm/3.4/arangodb3e.spec.in]: The exit was missing

**Pull request merged**